[Eugene Ang] iP

EangJS opened the pull request.

EangJS committed:
Add Increment: Level-0 (Duke Greetings)

EangJS committed:
Add Increment: Level-1 (Greet, Echo, Exit)

EangJS committed:
Added List and display list to the user

EangJS committed:
Added ability to mark as done and undone

EangJS committed:
Aligned to coding standard

EangJS committed:
Bug fixes for undisplayable char

EangJS committed:
Add todo,deadline,events and text testing

EangJS committed:
Refactor code

EangJS committed:
Correct typo error

EangJS committed:
Change formatting for packages

dendendenden04 commented on src/main/java/duke/command/UserCommandManager.java:
```diff
@@ -57,31 +59,31 @@ public static void handleUnknownCommand() {
-        default:
-            handleUnknownCommand();
-            break;
+            case "list":
```
Seems like an issue with indentation after the changes as the cases for the switch statement are indented afterwards. Can you verify?

EangJS replied:
Yup apparently i did not configure case indentation on my laptop ide

dendendenden04 commented on src/main/java/duke/command/TaskManager.java:
```diff
@@ -13,6 +19,7 @@ public TaskManager() {
      */
+
     public void addToList(String description) {
         Task task = new Task(description);
```
Inconsistent naming of variable, new task is named as "task" while for new deadlines and new event it is named as "event".

EangJS replied:
Oops my bad forgot to edit after copy pasting

dendendenden04 commented on src/main/java/duke/command/TaskManager.java:
```diff
@@ -28,11 +35,12 @@ public void addToList(String description) {
         this.listSize += 1;
-        String eventDuration = event.getdueDate();
+        String eventDuration = event.getDueDate();
         System.out.println("Got it! Added \n"
```
Good job for wrapping long lines around.

dendendenden04 reviewed (commented):
Hello there.

dendendenden04 commented on src/main/java/duke/command/TaskManager.java:
```diff
@@ -0,0 +1,114 @@
+public class TaskManager {
+    private Task[] taskList;
+    private int listSize;
+    private static final int MAXLISTSIZE = 100;
```
Do you think that you should add underscore for this constant variable.

EangJS reviewed (commented):
Hi thanks for your review appreciated it

TJ-Hoo commented on src/main/java/duke/command/UserCommandManager.java:
```diff
@@ -57,31 +59,31 @@ public static void handleUnknownCommand() {
+                break;
+            default:
+                handleUnknownCommand();
+                break;
```
Switch case very easy to read 10/10 code quality

TJ-Hoo commented on src/main/java/duke/task/Event.java:
```diff
@@ -1,3 +1,5 @@
+package duke.task;
```
Comments and short methods make code very easy to read and understand

TJ-Hoo commented on src/main/java/duke/command/TaskManager.java:
```diff
@@ -28,11 +35,12 @@ public void addToList(String description) {
+        String eventDuration = event.getDueDate();
         System.out.println("Got it! Added \n"
                 + "[D][ ] " + description + eventDuration + "\n"
                 + "to the list.");
```
Good job avoiding complicated expressions

TJ-Hoo commented on src/main/java/duke/command/TaskManager.java:
```diff
@@ -1,3 +1,9 @@
+
 public class TaskManager {
     private Task[] taskList;
     private int listSize;
```
Good job avoiding the usage of magic numbers

TJ-Hoo commented on src/main/java/duke/main/Duke.java:
```diff
@@ -1,3 +1,7 @@
+package duke.main;
```
Code is overall structured well and logically so is very easy to follow

EangJS committed:
Reformatted text

EangJS committed:
Add exception handling for array overflow and number formatting

-Handle error when user inputs string instead of a number
-Modify LIST_MAX_SIZE to 3 to test exception handling
-Update testcase

EangJS committed:
Merge branch 'branch-Level-5'

EangJS committed:
Modify error messages

EangJS committed:
Add delete function to remove task

- Changed from fixed length array to variable size with ArrayList<>
- Ability to remove task from a specified index
- Improve error message clarity

EangJS committed:
Add ability to save tasks to .json

EangJS committed:
Merge branch 'branch-Level-6'

EangJS committed:
Merge branch Level 6 and 7

EangJS committed:
Modify json structre

EangJS committed:
minor bug fixes regarding exception handling

EangJS committed:
Fix bug in which delete operator does not update .json

EangJS committed:
Refactor command package

EangJS committed:
Refactor storage and ui

EangJS committed:
Refactor Ui and input

EangJS committed:
Merge branch 'branch-A-MoreOOP'

EangJS committed:
Remove redundancy

EangJS committed:
Add date and time

EangJS committed:
Add sort and save

EangJS committed:
Merge pull request #1 from EangJS/branch-Level-8

Add sort and save

EangJS committed:
Add find task feature

EangJS committed:
Merge pull request #3 from EangJS/branch-Level-9

Add find task feature

EangJS committed:
Add javadoc comments to the code

EangJS committed:
Merge pull request #4 from EangJS/branch-A-JavaDoc

Add javadoc comments to the code

EangJS committed:
Remove redundant libs

EangJS committed:
Add user guide

EangJS committed:
Merge branch 'branch-A-UserGuide'

EangJS committed:
Update README.md

EangJS committed:
Update README.md

EangJS committed:
Update README.md

EangJS committed:
Update README.md

EangJS committed:
Rename Adding tasks.md to AddingTasks.md

EangJS committed:
Update README.md

EangJS committed:
Add links to pages

EangJS committed:
Add config.yaml

EangJS committed:
Add link to mainpage

EangJS committed:
Format pages

EangJS committed:
Format pages

EangJS committed:
Format pages and try new theme

EangJS committed:
Format page

EangJS committed:
Update _config.yml

EangJS committed:
Format page

EangJS committed:
Add find description in user guide

EangJS committed:
Add layouts

EangJS committed:
Add default.html

EangJS committed:
Add pre styles

EangJS committed:
Update default.html

EangJS committed:
Update default.html

EangJS committed:
Add navbar

EangJS committed:
Update default.html

EangJS committed:
Fix Website Bugs

EangJS committed:
Edit code block css style

EangJS committed:
Minor fomatting

EangJS committed:
Merge branch 'master' of github.com:EangJS/ip

chinhan99 commented on src/main/java/duke/command/DeleteFromList.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * Handles the delete command to remove task from list
+ */
+public class DeleteFromList extends Command {
```
May want to rename this class for consistency with other methods such as MarkCommand or ExitCommand

chinhan99 commented on src/main/java/duke/command/AddDeadlineToList.java:
```diff
@@ -0,0 +1,75 @@
+    private Deadline newTask;
+
+    /**
+     * preprocess userInput to create new deadline object
```
Do check for capitalisation and uses of periods in comments. 


https://se-education.org/guides/conventions/java/basic.html#comments

chinhan99 commented on src/main/java/duke/command/AddEventToList.java:
```diff
@@ -0,0 +1,78 @@
+            String startString = userInputArray[1];
+            String endString = userInputArray[2];
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+            try {
```
This try-catch might be redundant as it could be covered by the encompassing try-catch statement in the same method.

chinhan99 commented on src/main/java/duke/command/AddTodoToList.java:
```diff
@@ -0,0 +1,44 @@
+    private Task newTask;
+
+    /**
+     * create a new todo object
```
Do check on capitalisation and use of periods for comments

chinhan99 commented on src/main/java/duke/command/UserCommandManager.java:
```diff
@@ -0,0 +1,63 @@
+
+        Command command;
+        switch (userCommands[0]) {
+        case "bye":
+            command = new ExitCommand();
+            break;
+        case "list":
+        case "ls":
+            command = new ListTasks();
+            break;
+        case "list-sorted":
+        case "ls-s":
+            command = new PrintSortedTasks();
+            break;
+        case "find":
+            command = new FindTask(userCommands[1]);
+            break;
+        case "sort":
+            command = new SortTasks();
+            break;
+        case "mark":
+            command = new MarkCommand(userCommands[1]);
+            break;
+        case "unmark":
+            command = new UnmarkCommand(userCommands[1]);
+            break;
+        case "todo":
+            command = new AddTodoToList(userCommands[1]);
+            break;
+        case "deadline":
+            command = new AddDeadlineToList(userCommands[1]);
+            break;
+        case "event":
+            command = new AddEventToList(userCommands[1]);
+            break;
+        case "delete":
+            command = new DeleteFromList(userCommands[1]);
```
Good abstraction of submethods

chinhan99 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+     */
+    @Override
+    public String getTaskType() {
+        return "[D]";
```
A constant could be used to replace these strings.

chinhan99 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,94 @@
+        this.task = task;
+        this.isDone = false;
+        this.type = "todo";
+        this.endTime = LocalDateTime.of(0, 1, 1, 0, 0);
```
May want to consider converting the 0s and 1s to constants; so its easier for reviewers to understand your code

chinhan99 commented on text-ui-test/duke.json:
```diff
@@ -0,0 +1,41 @@
+{
+  "taskList": [
+    {
+      "type": "todo",
+      "task": " Buy milk",
+      "isDone": false
+    },
+    {
+      "dueBy": " 17/02/2023",
+      "type": "Deadline",
+      "task": "bake a cake ",
+      "isDone": true
+    },
+    {
+      "startTime": " 16/02/2023 ",
+      "endTime": " 19/02/2023",
+      "type": "Event",
+      "task": "attend baking course ",
+      "isDone": true
```
Understand that you are using a json file to store your variables. Do check w the prof if it can be used.

EangJS replied:
Hi, I think prof approved the gson library in the forum https://github.com/nus-cs2113-AY2223S2/forum/issues/22, should be fine right?

chinhan99 replied:
Ah! Should be fine then !good work !

chinhan99 reviewed (commented):
Overall good attempt at IP. Can see that you have integrated outside-module elements into your code ! To be safe,  you might want to check with the prof regarding the use of json files for storage. 

Do consider using constants to replace magic strings ; and also write the method comments with reference to the coding standards.

EangJS committed:
Capitalize and punctuate javadoc

EangJS committed:
Remove magic strings

EangJS committed:
Format code

EangJS committed:
Remove magic string filepath

EangJS committed:
Update text-ui-test

EangJS committed:
Update text-ui-test
